### PR TITLE
DLS-8893: Create Required User Answers - Change Business Activity

### PR DESCRIPTION
**Pull request by drewmiley**

### CODE REVIEW

File: test/controllers/actions/RequiredUserAnswersForChangeActivitySpec.scala [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+      }
+      implicit val request: DataRequest[AnyContentAsEmpty.type] = dataRequest(userAnswers)
+      val res = requiredUserAnswers.returnMissingAnswers(requiredUserAnswers.journey)
+      res mustBe List(RequiredPage(PackAtBusinessAddressPage, List(
+        PreviousPage(AmountProducedPage, List(AmountProduced.enumerable.withName("small").get,
+          AmountProduced.enumerable.withName("none").get))(implicitly[Reads[AmountProduced]]),
+        PreviousPage(ContractPackingPage, List(true))(implicitly[Reads[Boolean]])))(implicitly[Reads[Boolean]]))
+    }
+//
+//    s"should return 1 item on the missing answer list when user changes a $StartDatePage required answer from CYA, " +
```
Comment (GinnyThomas) [inline test/controllers/actions/RequiredUserAnswersForChangeActivitySpec.scala:501]: please remove unneeded code

File: app/controllers/actions/RequiredUserAnswersForChangeActivity.scala [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    List(
+      List(RequiredPage(AmountProducedPage, List.empty)(implicitAmountProduced)),
+      pagesRequiredForThirdPartyPackagersPage.map(RequiredPage(ThirdPartyPackagersPage, _)(implicitBoolean)),
+      pagesRequiredForOperatePackagingSiteOwnBrandsPage.map(RequiredPage(OperatePackagingSiteOwnBrandsPage, _)(implicitBoolean)),
+      pagesRequiredForHowManyOperatePackagingSiteOwnBrandsPage.map(RequiredPage(HowManyOperatePackagingSiteOwnBrandsPage, _)(implicitBands)),
+      List(RequiredPage(ContractPackingPage, List.empty)(implicitBoolean)),
+      pagesRequiredForHowManyContractPackingPage.map(RequiredPage(HowManyContractPackingPage, _)(implicitBands)),
+      List(RequiredPage(ImportsPage, List.empty)(implicitBoolean)),
+      pagesRequiredForHowManyImportsPage.map(RequiredPage(HowManyImportsPage, _)(implicitBands)),
+      pagesRequiredForPackAtBusinessAddressPage.map(RequiredPage(PackAtBusinessAddressPage, _)(implicitBoolean)),
```
Comment (GinnyThomas) [inline app/controllers/actions/RequiredUserAnswersForChangeActivity.scala:113]: see note below regarding pack at business address page

File: app/controllers/actions/RequiredUserAnswersForChangeActivity.scala [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      pagesRequiredForPackagingSiteDetailsPage.map(RequiredPage(PackagingSiteDetailsPage, _)(implicitBoolean))
+    ).flatten
+  }
+}
```
Comment (GinnyThomas) [inline app/controllers/actions/RequiredUserAnswersForChangeActivity.scala:117]: warehouse details page??

**Review by GinnyThomas — CHANGES_REQUESTED** [review]

File: test/controllers/actions/RequiredUserAnswersForChangeActivitySpec.scala (hunk 1 above)
Comment (drewmiley) [reply test/controllers/actions/RequiredUserAnswersForChangeActivitySpec.scala]: Done

File: test/controllers/actions/RequiredUserAnswersForChangeActivitySpec.scala [hunk 4]
```diff
@@ -0,0 +1,631 @@
...
+  "returnMissingAnswers" - {
+    "should return all missing answers when user answers is empty" in {
+      implicit val dataRequest: DataRequest[AnyContentAsEmpty.type] = RequiredDataRequest(FakeRequest(), "", aSubscription, emptyUserAnswersForChangeActivity)
+      val res = requiredUserAnswers.returnMissingAnswers(requiredUserAnswers.journey)
+      res mustBe
+        List(
+          RequiredPage(AmountProducedPage, List.empty)(implicitly[Reads[AmountProduced]]),
+          RequiredPage(ContractPackingPage, List.empty)(implicitly[Reads[Boolean]]),
+          RequiredPage(ImportsPage, List.empty)(implicitly[Reads[Boolean]]),
+        )
```
[the remark this answers is not in the document]
Comment (drewmiley) [reply test/controllers/actions/RequiredUserAnswersForChangeActivitySpec.scala:416]: I think this is valid as whether or not these are required will depend on the AmountProduced, and ContractPacking pages. So when they are missing, then we cannot yet say the others are.

File: test/controllers/actions/RequiredUserAnswersForChangeActivitySpec.scala [hunk 5]
```diff
@@ -0,0 +1,631 @@
...
+
+      s"when all $LitresInBands are not required" in {
+        val userAnswers = {
+          emptyUserAnswersForChangeActivity
+            .set(AmountProducedPage, Small).success.value
+            .set(ThirdPartyPackagersPage, true).success.value
+            .set(OperatePackagingSiteOwnBrandsPage, false).success.value
+            .set(ContractPackingPage, false).success.value
+            .set(ImportsPage, false).success.value
+            .set(PackAtBusinessAddressPage, true).success.value
```
[the remark this answers is not in the document]
Comment (drewmiley) [reply test/controllers/actions/RequiredUserAnswersForChangeActivitySpec.scala]: Should be fixed

File: test/controllers/actions/RequiredUserAnswersForChangeActivitySpec.scala [hunk 6]
```diff
@@ -0,0 +1,631 @@
...
+        val userAnswers = {
+          emptyUserAnswersForChangeActivity
+            .set(AmountProducedPage, Large).success.value
+            .set(OperatePackagingSiteOwnBrandsPage, true).success.value
+            .set(HowManyOperatePackagingSiteOwnBrandsPage, LitresInBands(1, 1)).success.value
+            .set(ContractPackingPage, true).success.value
+            .set(HowManyContractPackingPage, LitresInBands(1, 1)).success.value
+            .set(ImportsPage, true).success.value
+            .set(HowManyImportsPage, LitresInBands(1, 1)).success.value
+            .set(PackAtBusinessAddressPage, true).success.value
```
[the remark this answers is not in the document]
Comment (drewmiley) [reply test/controllers/actions/RequiredUserAnswersForChangeActivitySpec.scala:56]: This is for all journeys - regardless of whether they are coming from CYA? In this case, I may put the guard on the PackAtBusinessAddress controller

**Comment by drewmiley** [comment]
WIP - Need to fix RequiredUserAnswers unit tests, and add redirect guard to PackAtBusinessAddress

**Review by GinnyThomas — APPROVED** [review]

**Pull request merged**